test/graphql/queries/system_query_test.rb

```diff
@@ -250,6 +250,74 @@ class SystemQueryTest < ActiveSupport::TestCase
     assert_equal 'Never', result['data']['system']['lastScanned']
   end
 
+  should 'include inventory groups of host' do
+    @host1 = FactoryBot.create(:host, org_id: @user.account.org_id, groups: '1234')
+
+    query = <<-GRAPHQL
+    query getSystem($inventoryId: String!) {
+        system(id: $inventoryId) {
+          groups
+        }
+    }
+    GRAPHQL
+
+    result = Schema.execute(
+      query,
+      variables: { inventoryId: @host1.id },
+      context: { current_user: @user }
+    )
+
+    assert_equal @host1.groups, result['data']['system']['groups']
+  end
+
+  should 'serve search by group name' do
+    @host1 = FactoryBot.create(:host, org_id: @user.account.org_id, groups: [{ name: 'testgroup' }])
```

Comment:
Please create a host that does not match the search criteria and assert if it is not there